pkg/agent/openflow/pod_connectivity.go

```diff
@@ -0,0 +1,136 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package openflow
+
+import (
+	"net"
+
+	"antrea.io/antrea/pkg/agent/config"
+	"antrea.io/antrea/pkg/agent/openflow/cookie"
+	binding "antrea.io/antrea/pkg/ovs/openflow"
+)
+
+type featurePodConnectivity struct {
+	cookieAllocator cookie.Allocator
+	ipProtocols     []binding.Protocol
+
+	nodeCachedFlows *flowCategoryCache
+	podCachedFlows  *flowCategoryCache
+	fixedFlows      []binding.Flow
+
+	gatewayIPs    map[binding.Protocol]net.IP
+	ctZones       map[binding.Protocol]int
+	localCIDRs    map[binding.Protocol]net.IPNet
+	nodeIPs       map[binding.Protocol]net.IP
+	nodeConfig    *config.NodeConfig
+	networkConfig *config.NetworkConfig
+
+	connectUplinkToBridge bool
+	enableMulticast       bool
+
+	category cookie.Category
+}
+
+func (f *featurePodConnectivity) getFeatureName() string {
+	return "PodConnectivity"
+}
+
+func newFeaturePodConnectivity(
+	cookieAllocator cookie.Allocator,
+	ipProtocols []binding.Protocol,
+	nodeConfig *config.NodeConfig,
+	networkConfig *config.NetworkConfig,
+	connectUplinkToBridge bool,
+	enableMulticast bool) *featurePodConnectivity {
+	ctZones := make(map[binding.Protocol]int)
+	gatewayIPs := make(map[binding.Protocol]net.IP)
+	localCIDRs := make(map[binding.Protocol]net.IPNet)
+	nodeIPs := make(map[binding.Protocol]net.IP)
+	for _, ipProtocol := range ipProtocols {
+		if ipProtocol == binding.ProtocolIP {
+			ctZones[ipProtocol] = CtZone
+			gatewayIPs[ipProtocol] = nodeConfig.GatewayConfig.IPv4
+			localCIDRs[ipProtocol] = *nodeConfig.PodIPv4CIDR
+			nodeIPs[ipProtocol] = nodeConfig.NodeIPv4Addr.IP
+		} else if ipProtocol == binding.ProtocolIPv6 {
+			ctZones[ipProtocol] = CtZoneV6
+			gatewayIPs[ipProtocol] = nodeConfig.GatewayConfig.IPv6
+			localCIDRs[ipProtocol] = *nodeConfig.PodIPv6CIDR
+			nodeIPs[ipProtocol] = nodeConfig.NodeIPv6Addr.IP
+		}
+	}
+
+	return &featurePodConnectivity{
+		cookieAllocator:       cookieAllocator,
+		ipProtocols:           ipProtocols,
+		nodeCachedFlows:       newFlowCategoryCache(),
+		podCachedFlows:        newFlowCategoryCache(),
+		gatewayIPs:            gatewayIPs,
+		ctZones:               ctZones,
+		localCIDRs:            localCIDRs,
+		nodeIPs:               nodeIPs,
+		nodeConfig:            nodeConfig,
+		networkConfig:         networkConfig,
+		connectUplinkToBridge: connectUplinkToBridge,
+		enableMulticast:       enableMulticast,
+		category:              cookie.PodConnectivity,
+	}
+}
+
+func (f *featurePodConnectivity) initFlows() []binding.Flow {
+	var flows []binding.Flow
+
+	for _, ipProtocol := range f.ipProtocols {
+		if ipProtocol == binding.ProtocolIPv6 {
+			flows = append(flows, f.ipv6Flows()...)
+		} else if ipProtocol == binding.ProtocolIP {
+			flows = append(flows, f.arpNormalFlow())
+			if f.connectUplinkToBridge {
+				flows = append(flows, f.arpResponderFlow(f.nodeConfig.GatewayConfig.IPv4, f.nodeConfig.GatewayConfig.MAC))
+			}
```

Comment:
> But it seems now it's installed even when connectUplinkToBridge is disabled. Is it expected?

I added an if condition. It is installed only when connectUplinkToBridge is enabled. @tnqn